src/components/stepforms/AggregateStepForm.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <div>
+    <step-form-title :title="title"></step-form-title>
+    <WidgetMultiselect
+      id="groupbyColumnsInput"
+      v-model="editedStep.on"
+      name="Group by:"
+      :options="columnNames"
+      @input="setSelectedColumns({ column: editedStep.on[0] })"
+      placeholder="Add columns"
+    ></WidgetMultiselect>
+    <WidgetList
+      addFieldName="Add aggregation"
+      id="toremove"
+      name="Aggregations:"
+      v-model="aggregations"
+      :defaultItem="defaultAggregation"
+      :widget="'widget-aggregation'"
+      :automatic-new-field="false"
+    ></WidgetList>
+    <step-form-buttonbar :errors="errors" :cancel="cancelEdition" :submit="submit"></step-form-buttonbar>
+  </div>
+</template>
+
+<script lang="ts">
+import { Prop } from 'vue-property-decorator';
+import { AggFunctionStep, AggregationStep } from '@/lib/steps';
+import aggregateSchema from '@/assets/schemas/aggregate-step__schema.json';
+import WidgetMultiselect from './WidgetMultiselect.vue';
+import WidgetList from './WidgetList.vue';
+import BaseStepForm from './StepForm.vue';
+import { StepFormComponent } from '@/components/formlib';
+
+@StepFormComponent({
+  vqbstep: 'aggregate',
+  name: 'aggregate-step-form',
+  components: {
+    WidgetList,
+    WidgetMultiselect,
+  },
+})
+export default class AggregateStepForm extends BaseStepForm<AggregationStep> {
+  @Prop({ type: Object, default: () => ({ name: 'aggregate', on: [], aggregations: [] }) })
+  initialStepValue!: AggregationStep;
+
+  @Prop({
+    type: Boolean,
+    default: true,
+  })
+  isStepCreation!: boolean;
+
+  readonly title: string = 'Delete Column Step';
+
+  editedStepModel = aggregateSchema;
+
+  get defaultAggregation() {
+    const agg = {
+      column: '',
+      newcolumn: '',
+      aggfunction: 'sum',
+    };
+    return agg as AggFunctionStep;
+  }
+
+  get aggregations() {
+    if (this.editedStep.aggregations.length) {
+      return this.editedStep.aggregations;
+    } else {
+      return [this.defaultAggregation];
+    }
+  }
+
+  set aggregations(newval) {
+    this.editedStep.aggregations = newval;
+  }
+
+  submit() {
+    /**
+     * If different aggregations have to be performed on the same column, add a suffix
+     * to the automatically generated newcolumn name
+     */
+    const newcolumnOccurences: { [prop: string]: number } = {};
+    this.editedStep.aggregations.forEach(agg => {
+      agg['newcolumn'] = agg['column'];
+      newcolumnOccurences[agg.newcolumn] = (newcolumnOccurences[agg.newcolumn] || 0) + 1;
+    });
+    this.editedStep.aggregations.map(agg => {
```

Comment:
there's no reason to use `.map` here. We don't actually build a new list from the first one. A simple `for...of` loops would do the job.